src/Database/LSMTree/Internal/UniqCounter.hs

```diff
@@ -23,16 +25,42 @@ uniqueToRunNumber :: Unique -> RunNumber
 uniqueToRunNumber = coerce
 
 -- | An atomic counter for producing 'Unique' values.
+newtype UniqCounter m = UniqCounter (PrimVar (PrimState m) Int)
+-- Note that it is safe to store the 64-bit value as an `Int` type.
+-- The README.md clear states that the use must provide a machine architecture
+-- with a 64-bit Int type.
 --
--- TODO: could we use a PrimVar here?
-newtype UniqCounter m = UniqCounter (StrictMVar m Word64)
+-- Additionally, it is safe to @unsafeCoerce@ between `Int` and `Word64` for
+-- the same reason, the use must guarantee a 2's compliment representation for
```

Comment:
Are there cases where ints are not represented as 2's complement? Does it depend on GHC, `gcc`, the OS, the machine architecture?